File: Back/Cliente.Dominio/Entidade/Cliente.cs
using Clientes.Dominio.ObjetosDeValor;

namespace Clientes.Dominio.Entidade
{
    public class Cliente
    {
        public Guid Id { get; private set; }
        public string Nome { get; private set; }
        public Documento Documento { get; private set; }
        public DateTime DataNascimento { get; private set; }
        public Telefone Telefone { get; private set; }
        public Email Email { get; private set; }
        public Endereco Endereco { get; private set; }
        public string InscricaoEstadual { get; private set; }
        public bool Isento { get; private set; }

        protected Cliente() { }

        public Cliente(string nome, Documento documento, DateTime dataNascimento,
                      Telefone telefone, Email email, Endereco endereco,
                      string inscricaoEstadual = null, bool isento = false)
        {
            Id = Guid.NewGuid();
            Nome = nome;
            Documento = documento;
            DataNascimento = dataNascimento;
            Telefone = telefone;
            Email = email;
            Endereco = endereco;
            InscricaoEstadual = inscricaoEstadual;
            Isento = isento;
        }

        public void AtualizarDados(string nome, DateTime dataNascimento,
                                 Telefone telefone, Endereco endereco,
                                 string inscricaoEstadual = null, bool isento = false)
        {
            Nome = nome;
            DataNascimento = dataNascimento;
            Telefone = telefone;
            Endereco = endereco;
            InscricaoEstadual = inscricaoEstadual;
            Isento = isento;
        }

    }
}

File: Front/cliente-app/src/app/cliente/models/cliente.ts
export enum TipoDocumento {
  CPF = 0,
  CNPJ = 1
}

export interface Cliente {
  id: string;
  nome: string;
  documento: string;
  tipoDocumento: TipoDocumento;
  dataNascimento: string;
  telefone: string;
  email: string;
  cep: string;
  logradouro: string;
  numero: string;
  bairro: string;
  cidade: string;
  estado: string;
  inscricaoEstadual: string;
  isento: boolean;
  removido?: boolean;
}

export class ApiResponse<T> {
  result: T;
  success: boolean;
  successMessage: string;
  statusCode: number;
  errors: any[];
}

export class PagedInfo {
  pageNumber: number;
  pageSize: number;
  totalPages: number;
  totalRecords: number;
}

export class ClientesResponse {
  clientes: Cliente[];
  pagedInfo: PagedInfo;
}